src/term/resugar.rs

```diff
@@ -236,16 +244,19 @@ impl<'a> Reader<'a> {
   fn resugar_list(term: &mut Term) -> Term {
     match term {
       // (LCons el tail)
-      Term::App { fun: box Term::App { fun: ctr, arg: el, .. }, arg: tail, .. } => {
+      Term::App {
+        fun: box Term::App { fun: box Term::Ref { def_name: ctr }, arg: el, .. },
+        arg: tail,
+        ..
+      } => {
         let tail = Self::resugar_list(tail);
-        if let Term::List { els: tail } = tail {
+        if ctr.as_str() == LCONS && let Term::List { els: tail } = tail {
           let mut els = vec![*el.clone()];
           els.extend(tail);
           Term::List { els }
         } else {
           // Make the constructor again
-          let fun = Term::App { tag: Tag::Static, fun: ctr.clone(), arg: el.clone() };
-          Term::App { tag: Tag::Static, fun: Box::new(fun), arg: Box::new(tail) }
+          Term::call(Term::Ref { def_name: ctr.clone() }, [*el.clone(), tail])
```

Comment:
We can just std::mem::take(term) here instead of cloning

Reply:
I guess we can, but I'm refactoring this right now, so I'll leave it like that in this PR.